app/models/package_manager/nu_get.rb

```diff
@@ -237,23 +237,27 @@ def self.parse_requirements(range)
     class ParseCanonicalNameFailedError < StandardError; end
 
     # Returns canonical casing for case-insensitive NuGet package names
+    # @return String, false, nil
     def self.fetch_canonical_nuget_name(name)
       base_url = "https://nuget.org/packages/"
       page = get_html("#{base_url}#{name}")
       og_url_element = page.css("meta[property='og:url']").first
 
-      # This is likely to be a transient error or a sign the package was removed.
-      # A spike could indicate something systemic, however.
-      unless og_url_element
+      if page.text.empty? # Request failed, likely temporarily
         StructuredLog.capture("FETCH_CANONICAL_NAME_FAILED", { platform: "nuget", name: name })
+        return nil
+      elsif og_url_element.nil?
+        # If we got a response, but don't find this element, it most likely means
+        # the project was removed upstream.
+        StructuredLog.capture("CANONICAL_NAME_ELEMENT_MISSING", { platform: "nuget", name: name })
         return false
```

Comment:
These return values are bit of a stretch but seemed sufficient for the need to distinguish these cases.